### PR TITLE
Update Dockerfile to build java packages/binaries inside the Dockerfile

### PR DESCRIPTION
**Pull request by NehaNaithani**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,56 @@
-FROM jpetazzo/dind
+FROM jpetazzo/dind AS builder
```
Comment (baghbidi) [inline Dockerfile]: use this image as base for builder: `iofog/ubuntu-16.04-java8 `

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,56 @@
...
 
 RUN cd /opt && \
```
Comment (baghbidi) [inline Dockerfile]: The base image already has Java... no need to install it here. Move this to line 50, after `FROM jpetazzo/dind`

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,56 @@
...
+ENV PATH $PATH:$GRADLE_HOME/bin
+
+VOLUME $GRADLE_USER_HOME
+
+RUN gradle build copyJar
+
+FROM jpetazzo/dind
+
+COPY --from=builder packaging/iofog-agent/etc /etc
+COPY --from=builder packaging/iofog-agent/usr/bin /usr/bin
```
Comment (baghbidi) [inline Dockerfile]: Not needed since you are copying the whole /usr folder at line 53

**Review by baghbidi — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (NehaNaithani) [reply Dockerfile]: updated

File: Dockerfile (hunk 3 above)
Comment (NehaNaithani) [reply Dockerfile]: updated

File: Dockerfile (hunk 2 above)
Comment (NehaNaithani) [reply Dockerfile]: updated

**Review by baghbidi — APPROVED** [review]

**Pull request merged**